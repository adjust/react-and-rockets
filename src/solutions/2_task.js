// Please implement your solution in this file

export const API_URL = "https://api.spacexdata.com/v3/launches/past";

export const useRocketsData = () => {};
