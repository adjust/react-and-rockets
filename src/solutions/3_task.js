// Please implement your solution in this file

export const LOADING_MESSAGE = "Rockets loading...";

const FILTER_PARAMS = {
  year: 2018,
  customerName: "NASA",
};

export const RocketItem = ({ rocket }) => <div>RocketItem</div>;

export const RocketsList = () => <div>RocketsList</div>;
