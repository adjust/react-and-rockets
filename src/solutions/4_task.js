// Please implement your solution in this file

const FILTER_PARAMS = {
  year: 2018,
  customerName: "NASA",
};

export const RocketsList = () => <div>RocketsList</div>;
