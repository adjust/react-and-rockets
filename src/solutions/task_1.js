// Please implement your solution in this file

export const prepareData = () => {};
