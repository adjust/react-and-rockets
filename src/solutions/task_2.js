// Please implement your solution in this file

export const RocketsList = () => <div>ğŸš€ React & Rockets ğŸš€</div>;
