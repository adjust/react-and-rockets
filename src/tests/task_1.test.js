import { prepareData } from '../solutions/task_1';
import data from './__fixtures__/data.json';

/* 
  PLEASE DO NOT MODIFY THIS FILE
*/

describe(`# prepareData`, () => {
  test.each`
    year    | customerName
    ${2008} | ${'SpaceX'}
    ${2010} | ${'SpaceX'}
    ${2018} | ${'NASA'}
    ${2019} | ${'Iridium'}
    ${2018} | ${'CRS'}
    ${2018} | ${'S'}
    ${2018} | ${'t'}
    ${2016} | ${'t'}
  `(
    `consolidates the data correctly for year $year and customer $customerName`,
    ({ year, customerName }) => {
      expect(prepareData({ year, customerName })(data)).toMatchSnapshot();
    }
  );
});
