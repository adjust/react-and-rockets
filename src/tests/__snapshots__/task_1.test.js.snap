// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# prepareData consolidates the data correctly for year 2008 and customer SpaceX 1`] = `
Array [
  Object {
    "flight_number": 4,
    "mission_name": "RatSat",
    "payloads_count": 1,
  },
]
`;

exports[`# prepareData consolidates the data correctly for year 2010 and customer SpaceX 1`] = `
Array [
  Object {
    "flight_number": 6,
    "mission_name": "Falcon 9 Test Flight",
    "payloads_count": 1,
  },
]
`;

exports[`# prepareData consolidates the data correctly for year 2016 and customer t 1`] = `
Array [
  Object {
    "flight_number": 31,
    "mission_name": "ABS-2A / Eutelsat 117W B",
    "payloads_count": 2,
  },
  Object {
    "flight_number": 33,
    "mission_name": "JCSAT-16",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 29,
    "mission_name": "JCSAT-2B",
    "payloads_count": 1,
  },
]
`;

exports[`# prepareData consolidates the data correctly for year 2018 and customer CRS 1`] = `
Array [
  Object {
    "flight_number": 72,
    "mission_name": "CRS-16",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 64,
    "mission_name": "CRS-15",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 59,
    "mission_name": "CRS-14",
    "payloads_count": 1,
  },
]
`;

exports[`# prepareData consolidates the data correctly for year 2018 and customer NASA 1`] = `
Array [
  Object {
    "flight_number": 62,
    "mission_name": "Iridium NEXT Mission 6",
    "payloads_count": 2,
  },
  Object {
    "flight_number": 72,
    "mission_name": "CRS-16",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 64,
    "mission_name": "CRS-15",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 60,
    "mission_name": "TESS",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 59,
    "mission_name": "CRS-14",
    "payloads_count": 1,
  },
]
`;

exports[`# prepareData consolidates the data correctly for year 2018 and customer S 1`] = `
Array [
  Object {
    "flight_number": 62,
    "mission_name": "Iridium NEXT Mission 6",
    "payloads_count": 2,
  },
  Object {
    "flight_number": 56,
    "mission_name": "Paz / Starlink Demo",
    "payloads_count": 2,
  },
  Object {
    "flight_number": 73,
    "mission_name": "GPS III SV01",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 72,
    "mission_name": "CRS-16",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 71,
    "mission_name": "SSO-A",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 70,
    "mission_name": "Es’hail 2",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 64,
    "mission_name": "CRS-15",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 63,
    "mission_name": "SES-12",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 60,
    "mission_name": "TESS",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 59,
    "mission_name": "CRS-14",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 55,
    "mission_name": "Falcon Heavy Test Flight",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 54,
    "mission_name": "SES-16 / GovSat-1",
    "payloads_count": 1,
  },
]
`;

exports[`# prepareData consolidates the data correctly for year 2018 and customer t 1`] = `
Array [
  Object {
    "flight_number": 62,
    "mission_name": "Iridium NEXT Mission 6",
    "payloads_count": 2,
  },
  Object {
    "flight_number": 71,
    "mission_name": "SSO-A",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 70,
    "mission_name": "Es’hail 2",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 68,
    "mission_name": "Telstar 18V",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 66,
    "mission_name": "Iridium NEXT Mission 7",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 65,
    "mission_name": "Telstar 19V",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 58,
    "mission_name": "Iridium NEXT Mission 5",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 57,
    "mission_name": "Hispasat 30W-6",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 54,
    "mission_name": "SES-16 / GovSat-1",
    "payloads_count": 1,
  },
  Object {
    "flight_number": 53,
    "mission_name": "ZUMA",
    "payloads_count": 1,
  },
]
`;

exports[`# prepareData consolidates the data correctly for year 2019 and customer Iridium 1`] = `
Array [
  Object {
    "flight_number": 74,
    "mission_name": "Iridium NEXT Mission 8",
    "payloads_count": 1,
  },
]
`;
